src/boundaries.rs

```diff
@@ -1,5 +1,19 @@
 use std::collections::HashMap;
 
+use crate::geometry::{Coordinates, EdgeType, CornerType, Rectangle};
+
+fn _debug_log_to_file(message: String) {
+    use std::fs::OpenOptions;
+    use std::io::prelude::*;
+    let mut file = OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open("/tmp/mosaic-log.txt")
+        .unwrap();
+    file.write_all(message.as_bytes()).unwrap();
+    file.write_all("\n".as_bytes()).unwrap();
+}
```

Comment:
I'm not a terrible enough person to leave this in - will pick up the logging functionality